src/crypto/sign.js

```diff
@@ -0,0 +1,299 @@
+	};
+}
+
+/**
+ * @method decryptMessageWithSecret
+ * @param cipherHex
+ * @param nonce
+ * @param secret
+ * @param senderPublicKey
+ *
+ * @return {string}
+ */
+
+export function decryptMessageWithSecret(cipherHex, nonce, secret, senderPublicKey) {
+	const recipientPrivateKeyBytes = getRawPrivateAndPublicKeyFromSecret(secret).privateKey;
+	const convertedPrivateKey = convertPrivateKeyEd2Curve(recipientPrivateKeyBytes);
+	const senderPublicKeyBytes = hexToBuffer(senderPublicKey);
+	const convertedPublicKey = convertPublicKeyEd2Curve(senderPublicKeyBytes);
+	const cipherBytes = hexToBuffer(cipherHex);
+	const nonceBytes = hexToBuffer(nonce);
+
+	const decoded = naclInstance.crypto_box_open(
+		cipherBytes, nonceBytes, convertedPublicKey, convertedPrivateKey,
+	);
+
+	return naclInstance.decode_utf8(decoded);
+}
+
+/**
+ * @method signTransaction
+ * @param transaction Object
+ * @param secret Object
+ *
+ * @return {string}
+ */
+
+export function signTransaction(transaction, secret) {
+	const { privateKey } = getRawPrivateAndPublicKeyFromSecret(secret);
+	const transactionHash = getTransactionHash(transaction);
+	const signature = naclInstance.crypto_sign_detached(transactionHash, privateKey);
+	return bufferToHex(signature);
+}
+
+/**
+ * @method multiSignTransaction
+ * @param transaction Object
+ * @param secret Object
+ *
+ * @return {string}
+ */
+
+export function multiSignTransaction(transaction, secret) {
+	const transactionToSign = Object.assign({}, transaction);
+	delete transactionToSign.signature;
+	delete transactionToSign.signSignature;
+	const { privateKey } = getRawPrivateAndPublicKeyFromSecret(secret);
+	const bytes = getTransactionBytes(transactionToSign);
+	const transactionHash = getSha256Hash(bytes);
+	const signature = naclInstance.crypto_sign_detached(
+		transactionHash, privateKey,
+	);
+
+	return bufferToHex(signature);
+}
+
+/**
+ * @method verifyTransaction
+ * @param transaction Object
+ * @param secondPublicKey
+ *
+ * @return {boolean}
+ */
+
+export function verifyTransaction(transaction, secondPublicKey) {
+	const secondSignaturePresent = !!transaction.signSignature;
+	if (secondSignaturePresent && !secondPublicKey) {
+		throw new Error('Cannot verify signSignature without secondPublicKey.');
+	}
+
+	const transactionWithoutSignature = Object.assign({}, transaction);
+	if (secondSignaturePresent) delete transactionWithoutSignature.signSignature;
+	else delete transactionWithoutSignature.signature;
+
+	const transactionBytes = getTransactionBytes(transactionWithoutSignature);
+
+	const publicKey = secondSignaturePresent ? secondPublicKey : transaction.senderPublicKey;
+	const signature = secondSignaturePresent ? transaction.signSignature : transaction.signature;
+
+	const verified = naclInstance.crypto_sign_verify_detached(
+		hexToBuffer(signature), getSha256Hash(transactionBytes), hexToBuffer(publicKey),
+	);
+
+	return secondSignaturePresent ? verifyTransaction(transactionWithoutSignature) : verified;
```

Comment:
This verification style is inconsistent with the `verifyMessage` style: here a bad verification returns `false`, there it throws an error.

Reply:
Then we should change verfiyMessage to return false when not verified. But throw errors on not supported formats or wrong input.

Reply:
https://github.com/LiskHQ/lisk-js/issues/341